test/e2e/multihoming.go

```diff
@@ -2198,53 +2524,526 @@ func computeIPWithOffset(baseAddr string, increment int) (string, error) {
 	return netip.PrefixFrom(ip, addr.Bits()).String(), nil
 }
 
-// Given a node name and an offset, generateIPsFromNodePrimaryIfAddr returns an IPv4 and an IPv6 address
-// at the provided offset from the primary interface addresses found on the node.
-func generateIPsFromNodePrimaryIfAddr(cs clientset.Interface, nodeName string, offset int) ([]string, error) {
-	var newAddresses []string
-
-	node, err := cs.CoreV1().Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
+// buildRouteToHostSubnetViaExternalContainer returns ip route add commands to reach the host subnets via the provided gateway IPs
+func buildRouteToHostSubnetViaExternalContainer(cs clientset.Interface, f *framework.Framework, nodeName string, gwV4, gwV6, interfaceName string) ([]string, error) {
+	cmdTemplateV4 := "ip -4 route replace %s via " + gwV4 + " dev " + interfaceName
+	cmdTemplateV6 := "ip -6 route replace %s via " + gwV6 + " dev " + interfaceName
```

Comment:
I think using `fmt.Sprintf` is both more readable and more efficient (since you don't have to allocate and append strings 3 times. 

Your call though - this isn't production code, so it's not that important.